### ISSUE TITLE
templates with CQWP

### ISSUE THREAD

**Issue by Pieter-Veenstra** [SharePoint/PnP-Sites-Core#754]
#### Which PnP repository to report the issue?
- PnP Sites Core - https://github.com/OfficeDev/PnP-Sites-Core
- PnP PowerShell - https://github.com/OfficeDev/PnP-PowerShell
#### Category

[X] Bug
[ ] Enhancement
#### Environment

[X ] Office 365 / SharePoint Online
[ ] SharePoint 2016
[ ] SharePoint 2013
#### Expected or Desired Behavior

Exporting and importing pages with Content Query webparts should handle urls.
#### Observed Behavior

i've got a page with a CQWP when i look at the exported template I can see the following WebUrl:

<property name="WebUrl" type="string">~sitecollection/subsite/subsubsite</property>

once imported the site is looking for a list in the original site and therefore the web part is not working. This should probably use {site} instead.
#### Steps to Reproduce

Add CQWP on site
export site
apply template to new site.

**Comment by luismanez** [comment]
yup, it's a bug on the TokenizeWebpartXml (ObjectPageContents.cs). As the CSWP is already using the sitecollection token, none of the replacements tried in the method will work:

```c#
foreach (var list in lists)
            {
                xml = Regex.Replace(xml, list.Id.ToString(), string.Format("{{listid:{0}}}", list.Title), RegexOptions.IgnoreCase);
            }
            xml = Regex.Replace(xml, web.Id.ToString(), "{siteid}", RegexOptions.IgnoreCase);
            xml = Regex.Replace(xml, "(\"" + web.ServerRelativeUrl + ")(?!&)", "\"{site}", RegexOptions.IgnoreCase);
            xml = Regex.Replace(xml, "'" + web.ServerRelativeUrl, "'{site}", RegexOptions.IgnoreCase);
            xml = Regex.Replace(xml, ">" + web.ServerRelativeUrl, ">{site}", RegexOptions.IgnoreCase);
```

I'll try to do a fix and Pull Request when I got some time

**Comment by jansenbe** [comment]
Fix was implemented, closing this one.

**Comment by paulpascha** [comment]
It seems like this fix introducted a new bug #1066

**Comment by jakobnotseth** [comment]
In CQWP, it's important to keep the text ~sitecollection for example in WebUrl property, if it's replaced with empty as in a root site, the CQWP fails. If you need to fix this, you need some way of keeping ~site or ~sitecollection string and just replace the subweb string parts.
